core/src/main/scala/org/broadinstitute/dsp/workbench/welder/package.scala

```diff
@@ -129,17 +128,34 @@ package object welder {
       // We first try to read cache from local disk, if it exists, use it; if not, we read cache from gcs
       // Code for reading from disk can be deleted once we're positive that all user clusters are upgraded or when we no longer care.
       // This change is made on 3/26/2020
-      cacheFromDisk <- localCache[B](Paths.get(s"/work/.welder/${blobName.value.split("/")(1)}"))
-      loadedCache <- cacheFromDisk.fold {
-        storageAlg
-          .getBlob[List[B]](SourceUri.GsPath(stagingBucketName, blobName))
-          .last
-          .map(_.getOrElse(List.empty)) // The first time welder starts up, there won't be any existing cache, hence returning empty list
-      }(x => Stream.eval(IO.pure(x)))
+      traceId <- Stream.eval(ev.ask)
+      loadedCache <- sourceUri match {
+        case _: SourceUri.DataUri => Stream.eval(IO.raiseError(InvalidSourceURIException(traceId, "tried to get cachedresource for data uri", Map.empty)))
+        case GsPath(bucketName, blobName) =>
+          for {
+            cacheFromDisk <- localCache[B](Paths.get(s"/work/.welder/${blobName.value.split("/")(1)}"))
+            loadedCache <- cacheFromDisk.fold {
+              storageAlg
+                .getBlob[List[B]](sourceUri)
+                .last
+                .map(_.getOrElse(List.empty)) // The first time welder starts up, there won't be any existing cache, hence returning empty list
+            }(x => Stream.eval(IO.pure(x)))
+          } yield loadedCache
+        case SourceUri.AzurePath(containerName, blobName) =>
+          for {
+            cacheFromDisk <- localCache[B](Paths.get(s"/work/.welder/${blobName.value.split("/")(1)}"))
```

Comment:
```
localCache[B](Paths.get(s"/work/.welder/${blobName.value.split("/")(1)}"))
```
can be removed...it's not needed since this was just for backwards compatibility, and we don't need to worry about that for Azure right now